app/controllers/v1/profiles_controller.rb

```diff
@@ -18,6 +23,16 @@ def destroy
       render_json profile.destroy, status: :accepted
     end
 
+    def create
+      profile = Profile.new(profile_create_attributes).fill_from_parent
+
+      if profile.save
+        render_json profile, status: :created
+      else
+        render_error profile
```

Comment:
There's a scenario that looks like it could be handled here. 

1. POST contained a `business_objective` attribute "foo" - this BO doesn't exist in the current account
2. BO is created via `business_objective` 
3. Any error happens during `profile.save`, e.g: name already exists 

At this point, the "FOO" business objective will remain in the DB without being assigned to any profile.

Reply:
Good catch!